modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -155,36 +159,41 @@ public CompletableFuture<ReplicaResult> processRequest(ReplicaRequest request, S
         if (request instanceof PrimaryReplicaRequest) {
             var targetPrimaryReq = (PrimaryReplicaRequest) request;
 
+            CompletableFuture<Void> waitForActualStateFuture0 = waitForActualStateFuture.get();
+
             if (request instanceof WaitReplicaStateMessage) {
-                if (!waitForActualStateFuture.isDone()) {
-                    return processWaitReplicaStateMessage((WaitReplicaStateMessage) request)
+                if (waitForActualStateFuture0 == null) {
+                    if (waitForActualStateFuture.compareAndSet(null, new CompletableFuture<>())) {
+                        var waitReplicaStateMsg = (WaitReplicaStateMessage) request;
+
+                        return processWaitReplicaStateMessage(waitReplicaStateMsg)
+                                .thenApply(unused -> EMPTY_REPLICA_RESULT);
+                    } else {
+                        waitForActualStateFuture.get().thenApply(unused -> EMPTY_REPLICA_RESULT);
```

Comment:
What does this line of code do?

Reply:
This code sends the empty result as a response to WaitReplicaStateMessage. This branch happens if another WaitReplicaStateMessage has already started.